src/Network/Mattermost/Types.hs

```diff
@@ -49,38 +53,56 @@ maybeFail :: Parser a -> Parser (Maybe a)
 maybeFail p = (Just <$> p) <|> (return Nothing)
 
 -- | Creates a structure representing a TLS connection to the server.
-mkConnectionData :: Hostname -> Port -> ConnectionContext -> ConnectionData
-mkConnectionData host port ctx = ConnectionData
-  { cdHostname      = host
-  , cdPort          = port
-  , cdConnectionCtx = ctx
-  , cdAutoClose     = Yes
-  , cdToken         = Nothing
-  , cdLogger        = Nothing
-  , cdUseTLS        = True
+mkConnectionData :: Hostname -> Port -> Pool Connection -> ConnectionContext -> ConnectionData
+mkConnectionData host port pool ctx = ConnectionData
+  { cdHostname       = host
+  , cdPort           = port
+  , cdConnectionCtx  = ctx
+  , cdAutoClose      = No
+  , cdConnectionPool = pool
+  , cdToken          = Nothing
+  , cdLogger         = Nothing
+  , cdUseTLS         = True
   }
 
 -- | Plaintext HTTP instead of a TLS connection.
-mkConnectionDataInsecure :: Hostname -> Port -> ConnectionContext -> ConnectionData
-mkConnectionDataInsecure host port ctx = ConnectionData
-  { cdHostname      = host
-  , cdPort          = port
-  , cdConnectionCtx = ctx
-  , cdAutoClose     = Yes
-  , cdToken         = Nothing
-  , cdLogger        = Nothing
-  , cdUseTLS        = False
+mkConnectionDataInsecure :: Hostname -> Port ->  Pool Connection -> ConnectionContext -> ConnectionData
+mkConnectionDataInsecure host port pool ctx = ConnectionData
+  { cdHostname       = host
+  , cdPort           = port
+  , cdConnectionCtx  = ctx
+  , cdAutoClose      = No
+  , cdConnectionPool = pool
+  , cdToken          = Nothing
+  , cdLogger         = Nothing
+  , cdUseTLS         = False
   }
 
+stripesCount :: Int
+stripesCount = 1
+
+idleConnTimeout :: NominalDiffTime
+idleConnTimeout = 30
+
+maxConnCount :: Int
+maxConnCount = 5
+
```

Comment:
Can you please make a new record data type containing these settings, such as `data PoolSettings`, provide a `defaultPoolSettings` with these values, and then make that a parameter to `initConnectionData(Insecure)`?